yb-voyager/src/queryparser/unsupportedConstructs.go

```diff
@@ -315,8 +328,115 @@ func containsSystemColumnsInWhereClause(whereClause *pg_query.Node) bool {
+		}
+
+		xmlExpr := resTarget.Val.GetXmlExpr()
+		if xmlExpr != nil {
+			return true
+		}
+	}
+	return false
+}
+
+/*
+Query: SELECT * FROM xmltable(
+
+	$1
+	PASSING xmlparse(document $2)
+	COLUMNS name TEXT PATH $3
+
+) AS emp_data
+ParseTree: version:160001  stmts:{stmt:{select_stmt:{target_list:{res_target:{val:{column_ref:{fields:{a_star:{}}  location:7}}  location:7}}
+from_clause:{range_table_func:{docexpr:{xml_expr:{op:IS_XMLPARSE  args:{param_ref:{number:2  location:61}}
+args:{a_const:{boolval:{}  location:-1}}  xmloption:XMLOPTION_DOCUMENT  location:43}}  rowexpr:{param_ref:{number:1  location:28}}
+columns:{range_table_func_col:{colname:"name"  type_name:{names:{string:{sval:"text"}}  typemod:-1  location:82}  colexpr:{param_ref:{number:3  location:92}}  location:77}}  alias:{aliasname:"emp_data"}  location:14}}  limit_option:LIMIT_OPTION_DEFAULT  op:SETOP_NONE}}}
+*/
+func containsXmlFunctionsInFromClause(fromClause []*pg_query.Node) bool {
+	for _, fromItem := range fromClause {
+		// from clause is a subquery
+		if subselectNode, isSubselect := fromItem.Node.(*pg_query.Node_RangeSubselect); isSubselect {
+			subSelectStmt := subselectNode.RangeSubselect.Subquery.GetSelectStmt()
+			if subSelectStmt != nil {
+				// Recursively check for xml functions in the subquery's target list and FROM clause
+				if containsXmlFunctionsInTargetList(subSelectStmt.TargetList) {
+					return true
+				}
+				if containsXmlFunctionsInFromClause(subSelectStmt.FromClause) {
+					return true
+				}
+				if containsXmlFunctionsInWhereClause(subSelectStmt.WhereClause) {
+					return true
+				}
+			}
+		} else { // normal case
+			rangeTblFunc := fromItem.GetRangeTableFunc()
+			if rangeTblFunc == nil {
+				return false
+			}
+
+			xmlExpr := rangeTblFunc.Docexpr.GetXmlExpr()
+			if xmlExpr != nil {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+/*
+Query: SELECT id, name FROM employees WHERE xmlexists($1 PASSING BY VALUE data)
+ParseTree: version:160001  stmts:{stmt:{select_stmt:{target_list:{res_target:{val:{column_ref:{fields:{string:{sval:"id"}}  location:7}}  location:7}}
+target_list:{res_target:{val:{column_ref:{fields:{string:{sval:"name"}}  location:11}}  location:11}}
+from_clause:{range_var:{relname:"employees"  inh:true  relpersistence:"p"  location:22}}
+where_clause:{func_call:{funcname:{string:{sval:"pg_catalog"}}  funcname:{string:{sval:"xmlexists"}}
+args:{param_ref:{number:1  location:49}}  args:{column_ref:{fields:{string:{sval:"data"}}  location:69}}
+funcformat:COERCE_SQL_SYNTAX  location:39}}  limit_option:LIMIT_OPTION_DEFAULT  op:SETOP_NONE}}}
+*/
+func containsXmlFunctionsInWhereClause(whereClause *pg_query.Node) bool {
+	if whereClause == nil {
+		return false
+	}
+
+	if funcCallNode := whereClause.GetFuncCall(); funcCallNode != nil {
+		funcList := funcCallNode.Funcname
+		functionName := funcList[len(funcList)-1].GetString_().Sval
+		if slices.Contains(xmlFunctions, functionName) {
+			return true
+		}
+	}
+
```

Comment:
Thinking loud, can there be any case in which where clause can have the `XmlExpr` way as in the above target and from clause?

Reply:
yes @priyanshi-yb , one case can be - WHERE clause having a subquery

Reply:
But let me cover those later, need to think it through.

Reply:
okay